plugins/pagerduty/komand_pagerduty/actions/send_trigger_event/action.py

```diff
@@ -16,24 +16,34 @@ def __init__(self):
 
     def run(self, params={}):
         """Trigger event"""
-        ev = pypd.Event.create(
-            data={
-                "service_key": params["service_key"],
-                "event_type": "trigger",
-                "description": params["description"],
-                "contexts": params.get("contexts"),
-                "details": params.get("details"),
-                "client": params.get("client"),
-                "client_url": params.get("client_url"),
-            }
-        )
 
-        return ev
+        # required
+        email = params.get(Input.EMAIL)
+        title = params.get(Input.TITLE)
+        service = params.get(Input.SERVICE, {})
 
-    def test(self):
-        """Test event"""
-        return {
-            "incident_key": "aebdf1be9793454e86c0f0079820f32f",
-            "status": "success",
-            "message": "Event processed",
+        # optional
+        dict_of_optional_fields = {
+            "urgency": params.get(Input.URGENCY, ""),
+            "incident_key": params.get(Input.INCIDENT_KEY, ""),
+            "priority": params.get(Input.PRIORITY, {}),
+            "escalation_policy": params.get(Input.ESCALATION_POLICY, {}),
+            "conference_bridge": params.get(Input.CONFERENCE_BRIDGE, {}),
+            "body": params.get(Input.BODY, {}),
+            "assignments": params.get(Input.ASSIGNMENTS, []),
         }
+
+        if params.get("escalation_policy", {}) and params.get("assignments", []):
```

Comment:
Can we declare these two inputs as variables at the top then call the variable names in both the `dict_of_optional_fields` and this `if` statement

Reply:
this has been updated